packages/plugin-ext/src/main/browser/view/view-registry.ts

```diff
@@ -37,85 +35,51 @@ export class ViewRegistry {
     @inject(FrontendApplicationStateService)
     protected applicationStateService: FrontendApplicationStateService;
 
-    private containers: ViewContainerRegistry[] = new Array();
-
     private containersWidgets: Map<string, ViewsContainerWidget> = new Map<string, ViewsContainerWidget>();
+    private pending: Promise<void>;
 
-    private treeViewWidgets: Map<string, TreeViewWidget> = new Map<string, TreeViewWidget>();
-
-    @postConstruct()
-    init() {
-        this.applicationStateService.reachedState('ready').then(() => {
-            this.showContainers();
-            this.showTreeViewWidgets();
+    registerViewContainer(location: string, viewContainer: ViewContainer) {
+        const widget = new ViewsContainerWidget(viewContainer);
+        this.containersWidgets.set(viewContainer.id, widget);
+        if (!this.pending) {
+            this.pending = this.pendingReady();
+        }
```

Comment:
I think we can delete `pendingReady()` method and use `this.applicationStateService.reachedState('ready')` directly.
When the state is already `ready`, it will return resolved promise. If the state still not `ready`, it will return a promise, which will be resolved when the state become `ready`.
So, the code become shorter.
```
        if (!this.pending) {
            this.pending = this.applicationStateService.reachedState('ready');
        }
```

Reply:
ok. Done

Reply:
To me these lines should be refactored a bit. From an external point of view it is hard to understand what is `this.pending`. Maybe rename it ... but just `pending` doesn't mean anything to me.
Also, what if state 'ready' is already reached the first time ?

Reply:
Frontend application startup consists of the following steps:
- start frontend contributions
- attach the application shell to the host element
- initialize the application shell layout
- reveal the application shell if it was hidden by a startup indicator (state 'ready')

If state 'ready' is already reached the first time will be returned a new promise (deferred.promise)
https://github.com/theia-ide/theia/blob/master/packages/core/src/browser/frontend-application-state.ts#L61
https://github.com/theia-ide/theia/blob/master/packages/core/src/common/promise-util.ts#L26

Reply:
Done